### PR TITLE
add minizip

### LINKED ISSUE #23964: Package request: minizip
### Package name

minizip

### Package version

Newest

### Package website

 https://github.com/zlib-ng/minizip-ng
 

### Package availability

Github Releases

### Additional comments

This package is needed for conda-forge/freexl-feedstock#29  and would allow unvendoring within conda-forge/libkml-feedstock#2

I'll give this a swing, but I suspect others could do a better job :)

cc: @ocefpaf 

### Package is not available

- [X] The package is not available on conda-forge.

### No previous issues or open PRs

- [X] No previous issue exists and no PR has been opened.

### PR DESCRIPTION
**Pull request by akrherz**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
-  If static libraries are linked in, the license of the static library is packaged.
- [ ] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

closes #23964

### CODE REVIEW

**Comment by akrherz** [comment]
:wave: @ocefpaf , does this look like what you suggested for me to do?  I didn't "split" this as the `minizip` two binaries are rather small

I could use some kind help though from @conda-forge/help-c-cpp as I can't resolve what is plaguing the windows build and I likely have other things fouled up too :(  TIA and sorry

File: recipes/minizip/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+{% set version = "4.0.1" %}
+{% set sha256 = "63e47a2b4dbac0da501f43f4da74f118dfb3ef0dee0ffbbe89428271002260f8" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/zlib-ng/minizip-ng/archive/{{ version }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/minizip/meta.yaml:11]: Minor suggestion but we don't need this jinja variable.

```suggestion

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://github.com/zlib-ng/minizip-ng/archive/{{ version }}/{{ name }}-{{ version }}.tar.gz
  sha256: 63e47a2b4dbac0da501f43f4da74f118dfb3ef0dee0ffbbe89428271002260f8
```

File: recipes/minizip/build.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  -DBUILD_SHARED_LIBS=ON \
+  -DCMAKE_BUILD_TYPE=Release \
+  -DMZ_BUILD_TESTS=ON \
+  -DMZ_LIBCOMP=OFF \
+  -DMZ_ZLIB=ON \
+  -DCMAKE_INSTALL_INCLUDEDIR=include/minizip \
+  -DMZ_FORCE_FETCH_LIBS=OFF
+
+cmake --build build
+
```
Comment (ocefpaf) [inline recipes/minizip/build.sh:15]: I don't recall of they run test in that spec file I posted. Do we have a ctest or make check to run here?

**Comment by ocefpaf** [comment]
Windows is failing with:

```
CMake Error at cmake_install.cmake:130 (file):
  file INSTALL cannot find
  "C:/bld/minizip_1694612548479/work/Release/minigzip.exe": File exists
```

Not sure why it is trying to write that file again :-/

Maybe you don't need  `cmake --install . --config Release` there.

**Comment by akrherz** [comment]
> Windows is failing

Aren't all the linking fatal errors above that error of concern?

**Comment by ocefpaf** [comment]
> Aren't all the linking fatal errors above that error of concern?

I did not look at them :grimacing: I started from the last line ;-p

**Comment by akrherz** [comment]
For windows, I have been consulting [upstream's GHA](https://github.com/zlib-ng/minizip-ng/blob/master/.github/workflows/build.yml) and it has `MZ_FORCE_FETCH_LIBS=ON`, which we definitely do not want here (I believe). hmmm

**Comment by ocefpaf** [comment]
@isuruf we could use some help with Windows here :grimacing:

**Comment by akrherz** [comment]
@ocefpaf , the last windows build was dangerously close to working...  but still, anything to push this over the finish line would be appreciated!

File: recipes/minizip/bld.bat [hunk 3]
```diff
@@ -0,0 +1,21 @@
+set CC=cl
+
+cmake -S . -B . ^
+  -DCMAKE_INSTALL_PREFIX="%PREFIX%" ^
+  -DCMAKE_INSTALL_LIBDIR="%PREFIX%\lib" ^
```
Comment (isuruf) [inline recipes/minizip/bld.bat:5]:
```suggestion
  -DCMAKE_INSTALL_PREFIX="%LIBRARY_PREFIX%" ^
  -DCMAKE_INSTALL_LIBDIR="%LIBRARY_PREFIX%\lib" ^
```

**Comment by ocefpaf** [comment]
> I had to remove windows ctest stuff :(

I'm on mobile but I'll check the logs later. We need to know what is failing to be sure we are not publishing a bad package.

Anyway, we need to merge this to get it tested as a dependency on freexl. Awesome work @akrherz!!

**Comment by akrherz** [comment]
> We need to know what is failing

To clarify, the tests were not failing, I could not build the project to get to the point of running tests.  I'll create an issue once the repo is created to track looking into that further.

**Comment by ocefpaf** [comment]
> I could not build the project to get to the point of running tests.

I'm not sure I follow.

**Comment by akrherz** [comment]
> I'm not sure I follow.

To run tests you need to set `-DMZ_BUILD_TESTS=ON -DMZ_BUILD_UNIT_TESTS=ON`.  Setting that causes the windows build to fail before it even attempts to run tests.

**Comment by ocefpaf** [comment]
> To run tests you need to set `-DMZ_BUILD_TESTS=ON -DMZ_BUILD_UNIT_TESTS=ON`. Setting that causes the windows build to fail before it even attempts to run tests.

Ah. But that is what i want to see in the logs :smile: 
We can do that in a PR in the feedstock. Let's merge this!
PS: I was wondering if it is related to gtest but I'm still on mobile and I cannot read Azure logs on my cellphone. Too tiny.

**Pull request merged**